src/index.js

```diff
@@ -45,48 +46,71 @@ async function updatePackageJSON(pkg, options) {
     console.log("package.json not found");
     process.exit(1);
   }
+  let updatedPkg = {};
 
   if (pkg.scripts) {
-    pkg.scripts = upgradeScripts(pkg.scripts);
+    updatedPkg.scripts = upgradeScripts(pkg.scripts);
 
-    if (Object.values(pkg.scripts).some(s => s.includes('babel-node'))) {
+    if (Object.values(updatedPkg.scripts).some(s => s.includes('babel-node'))) {
       if (pkg.devDependencies) {
-        pkg.devDependencies["@babel/node"] = getLatestVersion();
+        updatedPkg.devDependencies = {
+          "@babel/node": getLatestVersion()
+        };
       }
     }
   }
 
-
   if (pkg.devDependencies) {
-    pkg.devDependencies = sortKeys(upgradeDeps(
-      pkg.devDependencies,
+    updatedPkg.devDependencies = sortKeys(upgradeDeps(
+      Object.assign(
+        {},
+        pkg.devDependencies,
+        updatedPkg.devDependencies || {}
+      ),
       getLatestVersion(),
       options,
     ));
   }
 
   if (pkg.dependencies) {
-    pkg.dependencies = sortKeys(upgradeDeps(
+    updatedPkg.dependencies = sortKeys(upgradeDeps(
       pkg.dependencies,
       getLatestVersion(),
       options,
     ));
   }
 
-  return pkg;
+  return Object.assign({}, pkg, updatedPkg);
 }
 
 async function writePackageJSON(options) {
   let { pkg, path } = await readPkgUp({ normalize: false });
-
-  pkg = await updatePackageJSON(pkg, options);
+  let updatedPkg = await updatePackageJSON(pkg, options);
 
   if (pkg.babel) {
     console.log("Updating package.json 'babel' config");
-    pkg.babel = upgradeConfig(pkg.babel, options);
+    updatedPkg.babel = upgradeConfig(pkg.babel, options);
+  }
+  if (options.dryRun) {
```

Comment:
maybe this block should happen whether or not we're doing a dry run? eg, have dry run _only_ control  whether `writeJsonFile` gets called